### PR TITLE
fix: Fix `slice` length no longer allowing `None`

### LINKED ISSUE #17365: `Expr.slice` no longer allows `length = None`
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl

df = pl.DataFrame({"foo": [1, 2, 3]})

df.select(pl.all().slice(1))
# InvalidOperationError: length must be integral for slice, not null
```

### Log output

_No response_

### Issue description

Previously, if no `length` was provided, all following rows would be selected. 

> If set to `None`, all rows starting at the offset will be selected.

### Expected behavior

```python
>>> df.select(pl.all().slice(1, pl.len()))
shape: (2, 1)
┌─────┐
│ foo │
│ --- │
│ i64 │
╞═════╡
│ 2   │
│ 3   │
└─────┘
```

### Installed versions

<details>

```
--------Version info---------
Polars:               1.0.0
Index type:           UInt32
Platform:             macOS-13.6.1-arm64-arm-64bit
Python:               3.12.2 (main, Feb  6 2024, 20:19:44) [Clang 15.0.0 (clang-1500.1.0.2.5)]

----Optional dependencies----
adbc_driver_manager:  <not installed>
cloudpickle:          <not installed>
connectorx:           <not installed>
deltalake:            <not installed>
fastexcel:            <not installed>
fsspec:               <not installed>
gevent:               <not installed>
great_tables:         <not installed>
hvplot:               <not installed>
matplotlib:           <not installed>
nest_asyncio:         <not installed>
numpy:                1.26.4
openpyxl:             <not installed>
pandas:               2.2.1
pyarrow:              15.0.2
pydantic:             <not installed>
pyiceberg:            <not installed>
sqlalchemy:           <not installed>
torch:                <not installed>
xlsx2csv:             <not installed>
xlsxwriter:           <not installed>
```

</details>

### PR DESCRIPTION
**Pull request by sherlockbeard**
Fixes #17365

### CODE REVIEW

**Review by stinodego — APPROVED** [review]
Thanks for the fix! Looks correct to me.

File: crates/polars-plan/src/plans/conversion/type_coercion/mod.rs [hunk 1]
```diff
@@ -451,7 +451,7 @@ impl OptimizationRule for TypeCoercionRule {
                 polars_ensure!(offset_dtype.is_integer(), InvalidOperation: "offset must be integral for slice, not {}", offset_dtype);
                 let (_, length_dtype) =
                     unpack!(get_aexpr_and_type(expr_arena, length, &input_schema));
-                polars_ensure!(length_dtype.is_integer(), InvalidOperation: "length must be integral for slice, not {}", length_dtype);
+                polars_ensure!(length_dtype.is_integer() || length_dtype.is_null(), InvalidOperation: "length must be integral for slice, not {}", length_dtype);
```
Comment (ritchie46) [inline crates/polars-plan/src/plans/conversion/type_coercion/mod.rs:454]: I think you should negate the check. `!is_null`.

File: py-polars/tests/unit/expr/test_exprs.py [hunk 2]
```diff
@@ -747,3 +747,16 @@ def test_slice_rejects_non_integral() -> None:
 
     with pytest.raises(pl.exceptions.InvalidOperationError):
         df.select(pl.col("a").slice(pl.lit("1"), None)).collect()
+
+
+def test_slice() -> None:
```
Comment (ritchie46) [inline py-polars/tests/unit/expr/test_exprs.py:752]: Can you test what you want fixed? E.g. ensure that we raise on the invalid input.

**Review by ritchie46 — CHANGES_REQUESTED** [review]

**Comment by ritchie46** [comment]
Ah, nevermind. I now see the issue. We want `None`.

**Review by ritchie46 — APPROVED** [review]

**Pull request merged**